cupy/cuda/compiler.py

```diff
@@ -145,6 +145,17 @@ def _get_arch():
         return min(arch, _nvrtc_max_compute_capability)
 
 
+def _get_arch_for_options(arch=None, jitify=False):
+    # This is needed to differentiate between
+    # compute or sm depending on the CUDA version
+    # needed to ensure backwards compatibility with nvrtc
```

Comment:
```suggestion
    # NVRTC in CUDA 11.3+ generates PTX that cannot be run with the driver
    # version earlier than the one included in that CUDA version, as
    # documented in:
    # https://docs.nvidia.com/cuda/archive/11.3.0/nvrtc/index.html#versioning
    # Here we use `-arch=sm_*` instead of `-arch=compute_*` to directly
    # generate cubin (SASS) instead of PTX. See #5097 for details.
```